app/src/main/java/mozilla/lockbox/view/OnboardingFingerprintAuthFragment.kt

```diff
@@ -0,0 +1,88 @@
+package mozilla.lockbox.view
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.jakewharton.rxbinding2.view.clicks
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subjects.PublishSubject
+import kotlinx.android.synthetic.main.onboarding_biometric_unlock.view.*
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import mozilla.lockbox.R
+import mozilla.lockbox.presenter.OnboardingFingerprintAuthPresenter
+import mozilla.lockbox.presenter.OnboardingFingerprintView
+import mozilla.lockbox.model.FingerprintAuthCallback
+import mozilla.lockbox.support.Constant
+import mozilla.lockbox.view.OnboardingFingerprintAuthFragment.AuthCallback.OnAuth
+import mozilla.lockbox.view.OnboardingFingerprintAuthFragment.AuthCallback.OnError
+
+@ExperimentalCoroutinesApi
+class OnboardingFingerprintAuthFragment : Fragment(), OnboardingFingerprintView {
+    private val compositeDisposable = CompositeDisposable()
+
+    private val _authCallback = PublishSubject.create<AuthCallback>()
+    override val authCallback: Observable<AuthCallback> get() = _authCallback
+
+    override val onDismiss: Observable<Unit>
+        get() = view!!.skipOnboarding.clicks()
+
+    private var isEnablingDismissed: Boolean = true
+
+    sealed class AuthCallback : FingerprintAuthCallback {
+        object OnAuth : AuthCallback()
+        object OnError : AuthCallback()
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        presenter = OnboardingFingerprintAuthPresenter(this)
+        return inflater.inflate(R.layout.onboarding_biometric_unlock, container, false)
+    }
+
+    override fun onSucceeded() {
+        view!!.sensorDescription.setTextColor(resources.getColor(R.color.green, null))
+        view!!.sensorDescription.text = getString(R.string.fingerprint_success)
+        view!!.iconFingerprint.setImageResource(R.drawable.ic_fingerprint_success)
+
+        view!!.sensorDescription.removeCallbacks(resetErrorTextRunnable)
+        view!!.iconFingerprint.postDelayed({
+            _authCallback.onNext(OnAuth)
+            isEnablingDismissed = false
+        }, Constant.FingerprintTimeouts.SUCCESS_DELAY_MILLIS)
+    }
+
+    override fun onFailed(error: String?) {
```

Comment:
This should be `@StringRes Int`.

Reply:
[I'm using the same logic that Ioana used in her Fingerprint dialog here]

This seems to be a string/charsequence because we are using this override method in the `FingerprintStore` which takes a charsequence:
```
inner class AuthenticationCallback(val context: Context) : FingerprintManager.AuthenticationCallback() {
  override fun onAuthenticationError(errorCode: Int, errString: CharSequence?) {
   super.onAuthenticationError(errorCode, errString)
   ... }
```